src/ros_user_id_password_authentication.cpp

```diff
@@ -0,0 +1,125 @@
+/*!
+ * \file ros_mac_authentication.cpp
+ * \brief Provides authentication via a message authentication codes (MAC).
+ *
+ * This node provides a service call that can be used to authenticate a user to use the ROS server.
+ * The node relies on a hash string that is made up of several pieces of information and hashed
+ * using the SHA-1 algorithm. This node is based on the best-practice method of a message
+ * authentication codes (MAC).
+ *
+ * \author Russell Toris, WPI - rctoris@wpi.edu
+ * \date March 4, 2013
+ */
+
+#include <fstream>
+#include <openssl/sha.h>
+#include <ros/ros.h>
+#include <rosauth/Authentication.h>
+#include <rosauth/UserIdPasswordAuthentication.h>
+#include <rosauth/md5.h>
+#include <sstream>
+#include <string>
+
+using namespace std;
+using namespace ros;
+
+/*!
+ * \def SECRET_FILE_USERS
+ * The ROS parameter name for the file that contains the users and passwords. We do not store the actual
+ * string in the parameter server as the parameter server itself may not be secure.
+ */
+#define SECRET_FILE_USERS "/rosauth/ros_mac_authentication/secret_file_users_location"
```

Comment:
shouldn't this be ```ros_user_id_password_authentication``` instead of ```/rosauth/ros_mac_authentication``` ? Otherwise, it is inconsistent with the launch file below.